xbmc/network/EventPacket.h

```diff
@@ -194,61 +196,31 @@ namespace EVENTPACKET
   class CEventPacket
   {
   public:
-    CEventPacket()
-    {
-      m_bValid = false;
-      m_iSeq = 0;
-      m_iTotalPackets = 0;
-      m_pPayload = NULL;
-      m_iPayloadSize = 0;
-      m_iClientToken = 0;
-      m_cMajVer = '0';
-      m_cMinVer = '0';
-      m_eType = PT_LAST;
-    }
-
-    CEventPacket(int datasize, const void* data)
-    {
-      m_bValid = false;
-      m_iSeq = 0;
-      m_iTotalPackets = 0;
-      m_pPayload = NULL;
-      m_iPayloadSize = 0;
-      m_iClientToken = 0;
-      m_cMajVer = '0';
-      m_cMinVer = '0';
-      m_eType = PT_LAST;
-
-      Parse(datasize, data);
-    }
-
-    virtual      ~CEventPacket() { free(m_pPayload); }
+    CEventPacket() = default;
+
+    explicit CEventPacket(int datasize, const void* data) { Parse(datasize, data); }
+
+    virtual ~CEventPacket() = default;
     virtual bool Parse(int datasize, const void *data);
     bool         IsValid() const { return m_bValid; }
     PacketType   Type() const { return m_eType; }
     unsigned int Size() const { return m_iTotalPackets; }
     unsigned int Sequence() const { return m_iSeq; }
-    void*        Payload() { return m_pPayload; }
-    unsigned int PayloadSize() const { return m_iPayloadSize; }
+    uint8_t* Payload() { return m_pPayload.data(); }
+    unsigned int PayloadSize() const { return m_pPayload.size(); }
     unsigned int ClientToken() const { return m_iClientToken; }
-    void         SetPayload(unsigned int psize, void *payload)
-    {
-      free(m_pPayload);
-      m_pPayload = payload;
-      m_iPayloadSize = psize;
-    }
+    void SetPayload(std::vector<uint8_t> payload) { m_pPayload = payload; }
```

Comment:
`m_pPayload = std::move(payload)`, and because std::move requires &lt;util> I would move this into the cpp.